File: server/gcc-common/src/main/java/org/ah/gcc/virtualrover/message/ServerRequestScreenshotMessage.java
package org.ah.gcc.virtualrover.message;

import org.ah.themvsus.engine.common.message.Message;
import org.ah.themvsus.engine.common.message.MessageCode;
import org.ah.themvsus.engine.common.message.MessageFactory;
import org.ah.themvsus.engine.common.transfer.Serializer;

import static org.ah.gcc.virtualrover.message.GCCMessageCode.ServerRequestScreenshot;

public class ServerRequestScreenshotMessage extends Message {

    public ServerRequestScreenshotMessage(MessageFactory messageFactory) {
        super(messageFactory);
    }

    @Override
    public MessageCode getType() { return ServerRequestScreenshot; }

    @Override
    public int size() {
        return super.size();
    }

    @Override
    protected void deserializeImpl(Serializer deserializer) {
        super.deserializeImpl(deserializer);
    }

    @Override
    protected void serializeImpl(Serializer serializer) {
    }
}

File: simulator/src/piwarssim/engine/message/ServerRequestScreenshotMessage.py
from piwarssim.engine.message.Message import Message


class ServerRequestScreenshotMessage(Message):
    def __init__(self, factory, message_type):
        super(ServerRequestScreenshotMessage, self).__init__(factory, message_type)

    def size(self):
        return super(ServerRequestScreenshotMessage, self).size()

    def deserialize_impl(self, deserializer):
        # super(FrameMessage, self).deserialize_impl(deserializer)
        pass

    def serialize_impl(self, serializer):
        # super(FrameMessage, self).serialize_impl(serializer)
        pass
